feature_engine/selection/shuffle_features.py

```diff
@@ -0,0 +1,213 @@
+            
+        else:
+            
+            model = RandomForestClassifier().fit(X,
+                                                 y)
+            y_pred = model.predict_proba(X)[:, 1]
+            model_performance = roc_auc_score(y,
+                                              y_pred)
+
+
+        
+        # dict to collect features and their performance_drift 
+        
+        self.features_performance_drifts_ = {}
+
+        # shuffle features and save feature performance drift into a dict
+
+        for feature in self.features_to_shuffle:
+
+        #  Create a copy of X
+            X_shuffled = X.copy()
+                                       
+            # shuffle individual feature
+            X_shuffled[feature] = X_shuffled[feature].sample(frac=1).reset_index(drop=True)
+            
+            # Use fitted model to calculate new target prediction with
+            # the shuffled feature.
+            # Then, calculate this model performance.
+            if (self.regression):
+                
+                shuffled_y_pred = model.predict(X_shuffled)
+                shuffled_model_performance = 1
+                
+            else:
+                
+                shuffled_y_pred = model.predict_proba(X_shuffled)[:, 1]
+                shuffled_model_performance = roc_auc_score(y,
+                                                           shuffled_y_pred)
+
+             
+            # Calculate drift in model performance after the feature has
+            # been shuffled.
+            drift = model_performance - shuffled_model_performance
+
+            # Save feature and its performance drift in the 
+            # features_performance_drifts_ attribute.
+            self.features_performance_drifts_[feature] = drift
+
+        return self
+
+    def transform(self, X):
+        """
+        
+        Updates the X dataframe with the new shuffled features.
+    
+        Parameters
+        ----------
+        
+        X: pandas dataframe of shape = [n_samples, n_features].
+            The input dataframe from which feature values will be shuffled.
+
+
+        Returns
+        -------
+
+        X_transformed: pandas dataframe of shape = [n_samples, n_features - len(dropped features)] 
+            Pandas dataframe with the selected features 
+        """
+
+        # check if fit is performed prior to transform
+
+        check_is_fitted(self)
+
+        # check if input is a dataframe
+
+        X = _is_dataframe(X)
+
+        # Create a list of the features to be kept depending on their 
+        # performance drift and the threshold and the ML problem.
+        
+        if (self.regression):
+            columns_to_drop = [feature for (feature, drift) in self.features_performance_drifts_.items() if drift >= self.threshold ]
+        else:
+            columns_to_drop = [feature for (feature, drift) in self.features_performance_drifts_.items() if drift <= self.threshold ]
+
```

Comment:
I think we should capture the selected features in an attribute in the fit method, and here we just drop  them